src/App.tsx

```diff
@@ -8,7 +8,54 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { filterTodos } from './helpers';
+import { Filters } from './types/Filters';
+
 export const App: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [todosFromServer, setTodosFromServer] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [filter, setFilter] = useState<Filters>(Filters.All);
+  const [query, setQuery] = useState('');
+
+  useEffect(() => {
+    getTodos()
+      .then(result => {
+        setTodosFromServer(result);
+        setIsLoading(false);
+      });
+  }, []);
+
+  const visibleTodos = filter !== Filters.All || query
+    ? filterTodos(todosFromServer, filter, query)
+    : todosFromServer;
```

Comment:
That's different abstraction levels. Can't you handle both cases in `filterTodos`?